support-frontend/app/services/MParticleClient.scala

```diff
@@ -84,18 +86,40 @@ class MParticleAuthClient(
       case Some(cachedToken) if cachedToken.expiresAt.isAfter(now) =>
         Future.successful(MParticleAccessToken(cachedToken.token))
```

Comment:
it's annoying that we need two atomic references with the same token in them, just to flag whether they are in date or not.

I can see that if a 401 response is very slow, we might have already refreshed the token and then it immediately gets cleared.

it feels like we should have one eventual atomic reference somehow together with a version number, and `invalidateAndRefreshToken` will only clear if it's the given version